### PR TITLE
Inventorymanagement to Development

### PR DESCRIPTION
**Pull request by dkwhite200**
--Inventory management (sprint 1)--

- Add CRUD module for inventory management. 
        - Create view
        - Edit view
        - List view
        - Details view
        - Created model for inventory
        - Delete, back and discontinue buttons to these pages

- Known Bugs
        - Urls without a route do no get 404 response 
        - too many back buttons with the header file and inventory managent pages

### CODE REVIEW

File: modules/inventorymanagements/client/config/inventorymanagements.client.routes.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  routeConfig.$inject = ['$stateProvider'];
+
+  function routeConfig($stateProvider) {
+    $stateProvider
+      .state('inventorymanagements', {
+        abstract: true,
+        url: '/inventorymanagements',
+        template: '<ui-view/>',
+        data: {
+          roles: ['user', 'admin'],
```
Comment (wcampbell09) [inline modules/inventorymanagements/client/config/inventorymanagements.client.routes.js]: I need this in my code

File: modules/inventorymanagements/client/controllers/inventorymanagements.client.controller.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    vm.authentication = Authentication;
+    vm.inventorymanagement = inventorymanagement;
+    vm.error = null;
+    vm.form = {};
+    vm.remove = remove;
+    vm.save = save;
+    vm.changeDCStatus = changeDCStatus;
+    vm.discBtn = discBtn;
+    vm.headSort = "tags";
```
Comment (wcampbell09) [inline modules/inventorymanagements/client/controllers/inventorymanagements.client.controller.js]: I need this too

Comment (wcampbell09) [inline modules/inventorymanagements/client/controllers/inventorymanagements.client.controller.js:22]: I need this line

File: modules/inventorymanagements/client/config/inventorymanagements.client.routes.js (hunk 1 above)
Comment (wcampbell09) [inline modules/inventorymanagements/client/config/inventorymanagements.client.routes.js:17]: I need this line of code

File: modules/inventorymanagements/client/views/list-inventorymanagements.client.view.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <thead class="thead-default">
+          <tr>
+            <th ng-click="vm.headSort = 'tags'">Tags</th>
+            <th ng-click="vm.headSort = 'upc'">UPC</th>
+            <th ng-click="vm.headSort = 'qty'">Quantity</th>
+            <th ng-click="vm.headSort = 'updated'">Last Updated</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr ng-repeat="inventorymanagement in vm.inventorymanagements  | filter : query | orderBy : vm.headSort"  ui-sref="inventorymanagements.view({ inventorymanagementId: inventorymanagement._id})">
```
Comment (wcampbell09) [inline modules/inventorymanagements/client/views/list-inventorymanagements.client.view.html:24]: I need this code too

**Review by eristow — APPROVED** [review]
Your code is good.

**Review by wcampbell09 — APPROVED** [review]
Code looks good

**Review by angryoctopode — APPROVED** [review]
Nicely done, much improved over your last failure.

**Review by AriTheGuitarMan — APPROVED** [review]
I like the part with the controllers. Nice styling. Forms good.

**Pull request merged**